### PR TITLE
Create API with CRUD functionality

### PR DESCRIPTION
**Pull request by lucyconklin**
API is working!

### CODE REVIEW

**Comment by akintner** [comment]
NICE!

**Comment by akintner** [comment]
I love this ` if !DeveloperCredential.find_by(api_key: key)` as the final option for that base_controller authenticate function. Way better than some of the other stuff we discussed!

**Comment by akintner** [comment]
closes #129

File: app/controllers/api/v1/base_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::BaseController < ApplicationController
+  before_action :authenticate_key
+
+  private
+
+    def authenticate_key
+      key = params[:api_key]
+      if !DeveloperCredential.find_by(api_key: key)
+        # byebug
```
Comment (CPowell23) [inline app/controllers/api/v1/base_controller.rb:9]: Byebug still in the controller

**Pull request merged**